### PR TITLE
Connection Interface Exposed & Refactored

### PR DESCRIPTION
**Pull request by mattcreager**
- goConnect has been retired as promises in core makes it redundant
- goUsers has been retired as promises in core makes it redundant
- The user model and factory have been retired
- Connection has been added, it's a very simple wrapper around connect
  that makes configuration via. Angular possible
- goConnection is also a regitered service, it can be used to access
  anything normally available on the connection object

Further Information in Issue #16

### CODE REVIEW

File: tests/connection.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    describe('configuration', function() {
+
+      it('extends the connection object with url and options', function() {
+        var goConnection = new Connection();
+
+        goConnection.set(url, opts);
+
+        assert.equal(goConnection._opts, opts);
+        assert.equal(goConnection._url, url);
+        assert(goConnection._configured);
```
Comment (ianlivingstone) [inline tests/connection.js:50]: You should assert more about the configuration then just the fact it has been set

File: tests/connection.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        get = sandbox.stub().returns('connection');
+        window.goinstant = {};
+        window.goinstant.connect = sandbox.stub().returns({ get: get });
+      });
+
+      it('connects to goinstant', function() {
+        goConnection.set(url, opts);
+        goConnection.$get();
+
+        assert(goConnection._connecting);
```
Comment (ianlivingstone) [inline tests/connection.js:68]: You're using boolean flags here to set the state of the world, why not a state variable that contains strings?

File: lib/connection.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+/**
+ * This method is invoked by Angulars dependency injection system and initiates
+ * the connection with GoInstant
+ */
+Connection.prototype.$get = function() {
+  if (!window.goinstant) {
+    throw new Error('GoAngular requires the GoInstant library.');
+  }
+
+  if (!this._connecting) {
```
Comment (ianlivingstone) [inline lib/connection.js:66]: Ok ignore my comment on the test, no need for a state machine here a simple boolean flag will do

File: lib/connection.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  }
+
+  return this;
+};
+
+/**
+ * Responsible for connecting to goinstant, assigns a promise to the private
+ * connection property
+ */
+Connection.prototype._connect = function() {
```
Comment (ianlivingstone) [inline lib/connection.js:77]: what if `_connect` gets called twice? what if connect fails?

File: lib/go_angular.js [hunk 5]
```diff
@@ -179,7 +182,7 @@ GoAngular.prototype._handleScopeChanges = _.debounce(function() {
  * @todo handle errors when no listeners are available
  */
 GoAngular.prototype.destroy = function destroy(cb) {
```
Comment (ianlivingstone) [inline lib/go_angular.js:184]: this doesn't return a promise?

File: tests/connection.js (hunk 1 above)
Comment (mattcreager) [inline tests/connection.js:50]: Just above that last assertion: _opts is asserted, and _url is asserted, those are the only other options!

File: tests/connection.js (hunk 2 above)
Comment (mattcreager) [inline tests/connection.js:68]: I considered it, but in the end, we only have two flags: configured and connecting!

File: lib/connection.js (hunk 3 above)
Comment (mattcreager) [inline lib/connection.js:66]: Ya, I did start with a state machine in the original, just cut, cut, cut in this... I'm trying to minimize dependencies and footprint.

File: lib/connection.js (hunk 4 above)
Comment (mattcreager) [inline lib/connection.js:77]: The boolean check above should prevent that, but the beauty of promises is it makes no difference, the promise will always return the same result, whether its been resolved or rejected! I (heart) promises!

File: lib/go_angular.js (hunk 5 above)
Comment (mattcreager) [inline lib/go_angular.js:184]: I didn't modify this in this PR, but don't worry, it's going to get a major refactor!

**Pull request closed without merging**